### PR TITLE
feat(tests): add enametoolong tests

### PR DESCRIPTION
**Pull request by musikid**

### CODE REVIEW

File: rust/src/tests/errors/enametoolong.rs [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+            );
+        }
+    };
+}
+
+pub(crate) use enametoolong_either_comp_test_case;
+
+/// Create a test case which asserts that the sycall
+/// returns `ENAMETOOLONG` if an entire pathname
+/// exceeded `{PATH_MAX}` characters.
```
Comment (asomers) [inline rust/src/tests/errors/enametoolong.rs:94]:
```suggestion
/// exceeds `{PATH_MAX}` characters.
```

File: rust/src/tests/errors/enametoolong.rs [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+                $syscall($( $($($before),* ,)? )? path $( $(, $($after),*)? )?)
+        });
+    };
+}
+
+pub(crate) use enametoolong_comp_test_case;
+
+/// Create a test case which asserts that the sycall
+/// returns `ENAMETOOLONG` if a component of either pathname
+/// exceeded `{NAME_MAX}` characters.
```
Comment (asomers) [inline rust/src/tests/errors/enametoolong.rs:58]:
```suggestion
/// exceeds `{NAME_MAX}` characters.
```

File: rust/src/tests/errors/enametoolong.rs [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+                $syscall($( $($($before),* ,)? )? path $( $(, $($after),*)? )?)
+        });
+    };
+}
+
+pub(crate) use enametoolong_path_test_case;
+
+/// Create a test case which asserts that the sycall
+/// returns `ENAMETOOLONG` if an entire pathname
+/// exceeded `{PATH_MAX}` characters.
```
Comment (asomers) [inline rust/src/tests/errors/enametoolong.rs:149]:
```suggestion
/// exceeds `{PATH_MAX}` characters.
```

File: rust/src/tests/mknod.rs [hunk 4]
```diff
@@ -188,5 +189,26 @@ fn create_old_new_device(ctx: &mut TestContext) {
...
+    fn mknod_block_wrapper(_: &mut TestContext, path: &Path) -> nix::Result<()> {
+        mknod(path, SFlag::S_IFBLK, Mode::empty(), 0)
+    }
+
+    fn mknod_char_wrapper(_: &mut TestContext, path: &Path) -> nix::Result<()> {
+        mknod(path, SFlag::S_IFCHR, Mode::empty(), 0)
+    }
+
+    // mknod/02.t
+    enametoolong_comp_test_case!(mknod, mknod_block_wrapper, mknod_char_wrapper; root);
```
Comment (asomers) [inline rust/src/tests/mknod.rs:213]: Really?  mknod needs root privileges in this case but not in the whole-path case?

Comment (musikid) [reply rust/src/tests/mknod.rs:213]: Oh no, that's a mistake on my part!

**Review by asomers — APPROVED** [review]

**Pull request merged**